content/post/kubestash-v2024.6.4/index.md

```diff
@@ -0,0 +1,258 @@
+      name: s3-storage
+    retentionPolicy:
+      name: demo-retention
+      namespace: demo
+  sessions:
+  - name: workload-backup
+    sessionHistoryLimit: 3
+    scheduler: 
+      schedule: "*/5 * * * *"
+      jobTemplate:
+        backoffLimit: 1
+    repositories:
+    - name: demo-storage
+      backend: s3-storage
+      directory: /sts
+      encryptionSecret:
+        name: encry-secret
+        namespace: demo
+    addon:
+      name: workload-addon
+      tasks:
+      - name: manifest-backup
+    retryConfig:
+      maxRetry: 2
+      delay: 1m
+```
+
+Here is the example of `RestoreSession` that restores the `StatefulSet` manifests:
+
+```yaml
+apiVersion: core.kubestash.com/v1alpha1
+kind: RestoreSession
+metadata:
+  name: sample-restore
+  namespace: demo
+spec:
+  manifestOptions:
+    restoreNamespace: dev
+  dataSource:
+    repository: demo-storage
+    snapshot: latest
+    encryptionSecret:
+      name: encry-secret
+      namespace: demo
+  addon:
+    name: workload-addon
+    tasks:
+    - name: manifest-restore
+```
+
+Here, we can configure in which namespace we want to restore our workload by providing the namespace in `spec.manifestOptions.restoreNamespace`.
+
+#### Unified Manifest and Data Recovery
+
+Now you can restore the manifests and data of workloads or databases (`MySQL`, `MariaDB`, `MongoDB`, `PostgreSQL`) by creating just one `RestoreSession`. KubeStash will deploy the workload or database in a cluster from the backed-up manifests and then restore data into it from the backed-up data. For this, you need to configure the manifest backup and data backup in the same session of a `BackupConfiguration`. This enables KubeStash to restore the manifests and data using a single `Snapshot`.
+
+Here is an example of a `BackupConfiguration` for backing up both the manifests and data of a `StatefulSet` in the same session:
+
+```yaml
+apiVersion: core.kubestash.com/v1alpha1
+kind: BackupConfiguration
+metadata:
+  name: sample-backup
+  namespace: demo
+spec:
+  target:
+    apiGroup: apps
+    kind: StatefulSet
+    name: sample-sts
+    namespace: demo
+  backends:
+  - name: s3-storage
+    storageRef:
+      namespace: demo
+      name: s3-storage
+    retentionPolicy:
+      name: demo-retention
+      namespace: demo
+  sessions:
+  - name: workload-backup
+    sessionHistoryLimit: 3
+    scheduler:
+      schedule: "*/5 * * * *"
+      jobTemplate:
+        backoffLimit: 1
+    repositories:
+    - name: demo-storage
+      backend: s3-storage
+      directory: /sts
+      encryptionSecret:
+        name: encry-secret
+        namespace: demo
+    addon:
+      name: workload-addon
+      tasks:
+      - name: manifest-backup
+      - name: logical-backup
+        params:
+          paths: /source/data
+          exclude: /source/data/tmp
+    retryConfig:
+      maxRetry: 2
+      delay: 1m
+```
+
+Here's an example of a `RestoreSession` that restores `StatefulSet` manifests and then restores its data:
```

Comment:
Here is an example of a RestoreSession that first redeploys the StatefulSet from the backed-up manifests and then restores the data into it: